components-compose/src/main/java/uk/gov/hmrc/components/compose/ui/theme/HmrcDevicePreviews.kt

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.ptcalc.common.compose.core
+
+import android.content.res.Configuration
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+
+/**
+ * Device previews for light and dark mode
+ */
+@Preview(
+    name = "Phone - Light",
+    device = Devices.PIXEL_4,
+    uiMode = Configuration.UI_MODE_NIGHT_NO,
+    showBackground = true,
+)
+@Preview(
+    name = "Phone - Dark",
+    device = Devices.PIXEL_4,
+    uiMode = Configuration.UI_MODE_NIGHT_YES,
+    showBackground = true,
+)
+@Preview(
+    name = "Tablet - Light",
+    device = Devices.PIXEL_C,
+    uiMode = Configuration.UI_MODE_NIGHT_NO,
+    showBackground = true,
+)
+@Preview(
+    name = "Tablet - Dark",
+    device = Devices.PIXEL_C,
+    uiMode = Configuration.UI_MODE_NIGHT_YES,
+    showBackground = true,
+)
+@Preview(
+    name = "Large Font mode",
+    uiMode = Configuration.UI_MODE_NIGHT_NO,
+    fontScale = 1.5f,
```

Comment:
I think it might be better to make this even bigger to act as a test for large text layouts.. Our biggest in our main codebase is a scale of 2f so let's go with that.